core/src/main/java/com/amplifyframework/auth/AuthException.java

```diff
@@ -191,4 +191,384 @@ public UnknownException(Throwable cause) {
             );
         }
     }
+
+    /**
+     * Exception thrown by AWS Cognito Auth.
+     */
+    public static class AWSCognitoAuthException extends AuthException {
```

Comment:
Since you define its subtypes inline in the `AWSCognitoAuthException`, I believe you can mark this exception (and all of its subtypes) as `final` classes.